scripts/sort-makefile-am-sources.pl

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/perl
+#
+## Copyright (C) 1996-2020 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+use strict;
+use warnings;
+
+my $current_source_section='';
+while (<>) {
+    if (m!^#!) {
+        print;
+        next;
+    }
+    if (/^(\S+_SOURCES)\s*=\s*\\$/) {
+        $current_source_section=$1;
+        print "$1 = \\\n";
+    } else {
+        print;
+        next;
+    }
+    # accumulate files and prep for sorting
+    my %files = ();
+    while (<>) {
+
+        chomp;
+        m!\s*(tests/stub_|tests/test)?(\S+)(\s+\\\s*)?$! || die "no parse";
+        my $prefix = (defined $1) ? $1 : '';
+        my $filename = (defined $2) ? $2 : '';
+        
+        print STDERR "WARNING: duplicate $prefix$filename ".
+            "detected in $current_source_section"
+            if (exists($files{"$filename.$prefix"}));
```

Comment:
Nit:

```suggestion
            if exists $files{"$filename.$prefix"};          
```

I do not insist on this readability improvement change.

Reply:
that's fine, I can add it. Doing that